### PR TITLE
Split paddle_capi_whole into paddle_nn_engine and paddle_layers two s…

### PR DESCRIPTION
**Pull request by hedaoyuan**
…tatic libraries.
Here is a use case https://github.com/PaddlePaddle/Mobile/pull/13.

### CODE REVIEW

File: paddle/capi/CMakeLists.txt [hunk 1]
```diff
@@ -38,10 +37,13 @@ set(PADDLE_CAPI_INFER_LIBS
     paddle_function
     paddle_gserver
     paddle_proto)
-
 cc_library(paddle_capi_whole DEPS paddle_capi ${PADDLE_CAPI_INFER_LIBS})
 
-# No shared library for iOS
+# Link the static library for inference
+cc_library(paddle_nn_engine DEPS paddle_capi paddle_utils paddle_parameter paddle_math paddle_cuda paddle_proto)
+cc_library(paddle_layers DEPS paddle_function paddle_gserver)
```
Comment (Xreki) [inline paddle/capi/CMakeLists.txt]: About the libraries' name, how about to name them `paddle_capi_engine` and `paddle_capi_layers`?

File: paddle/capi/CMakeLists.txt [hunk 2]
```diff
@@ -55,7 +57,7 @@ endif()
 install(FILES ${CAPI_HEADERS} DESTINATION include/paddle)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/config.h DESTINATION include/paddle)
 if(ANDROID)
-  install(TARGETS paddle_capi_whole paddle_capi_shared
+  install(TARGETS paddle_nn_engine paddle_layers paddle_capi_shared
```
Comment (Xreki) [inline paddle/capi/CMakeLists.txt]: Do we need to install `paddle_capi_whole` for a period of time?

File: paddle/capi/CMakeLists.txt (hunk 1 above)
Comment (hedaoyuan) [reply paddle/capi/CMakeLists.txt]: OK

Comment (hedaoyuan) [reply paddle/capi/CMakeLists.txt]: Done.

File: paddle/capi/CMakeLists.txt (hunk 2 above)
Comment (hedaoyuan) [reply paddle/capi/CMakeLists.txt]: Done.

File: paddle/capi/CMakeLists.txt [hunk 3]
```diff
@@ -28,20 +28,36 @@ add_style_check_target(paddle_capi ${CAPI_SOURCES} ${CAPI_HEADER}
...
+        paddle_utils
+        paddle_parameter
+        paddle_math
+        paddle_cuda
+        paddle_function
+        paddle_gserver
+        paddle_proto
+        paddle_pserver
+        paddle_network)
+    cc_library(paddle_capi_whole DEPS paddle_capi ${PADDLE_CAPI_INFER_LIBS})
```
Comment (Xreki) [inline paddle/capi/CMakeLists.txt]: `line 53` is the same as `line 41`. It can be moved out of the `if` statement.

Comment (hedaoyuan) [reply paddle/capi/CMakeLists.txt]: Done.

**Pull request merged**